src/app/leverage/provider.tsx

```diff
@@ -324,7 +324,7 @@ export function LeverageProvider(props: { children: any }) {
         isMinting,
         inputToken: token,
         outputToken,
-        network: token.chainId,
+        network: chainId,
```

Comment:
this may not even be necessary but might make more sense?

Reply:
hmm, do we run the risk of the `chainId` the user is currently connected to not having the token on the corresponding chain?

Reply:
thought about this and tested a bit - the list of tokens shown should be for the same `chainId` that the user is already on (and hence, `token.chainId` should match their current value), so the code will work the same either way